plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java

```diff
@@ -237,6 +238,39 @@ public void testPredictRemoteModel() throws IOException, InterruptedException {
         assertFalse(((String) responseMap.get("text")).isEmpty());
     }
 
+    public void testPredictWithAutoDeployAndTTL_RemoteModel() throws IOException, InterruptedException {
+        // Skip test if key is null
+        if (OPENAI_KEY == null) {
+            System.out.println("OPENAI_KEY is null");
+            return;
+        }
+        Response response = createConnector(completionModelConnectorEntity);
+        Map responseMap = parseResponseToMap(response);
+        String connectorId = (String) responseMap.get("connector_id");
+        response = registerRemoteModelWithTTL("openAI-GPT-3.5 completions", connectorId, 1);
+        responseMap = parseResponseToMap(response);
+        String modelId = (String) responseMap.get("model_id");
+        String predictInput = "{\n" + "  \"parameters\": {\n" + "      \"prompt\": \"Say this is a test\"\n" + "  }\n" + "}";
+        response = predictRemoteModel(modelId, predictInput);
+        responseMap = parseResponseToMap(response);
+        List responseList = (List) responseMap.get("inference_results");
+        responseMap = (Map) responseList.get(0);
+        responseList = (List) responseMap.get("output");
+        responseMap = (Map) responseList.get(0);
+        responseMap = (Map) responseMap.get("dataAsMap");
+        responseList = (List) responseMap.get("choices");
+        if (responseList == null) {
+            assertTrue(checkThrottlingOpenAI(responseMap));
+            return;
+        }
+        responseMap = (Map) responseList.get(0);
+        assertFalse(((String) responseMap.get("text")).isEmpty());
+
+        getModelProfile(modelId, verifyRemoteModelDeployed());
+        TimeUnit.SECONDS.sleep(71);
```

Comment:
Why do we have to wait 11 more seconds?

Reply:
auto undeploy is in the cron job which runs every 10 seconds in the IT. So the max time is ttl + 10.